### ISSUE TITLE
unnecessary-list-index-lookup is falsely being reported when there is a start index

### ISSUE THREAD

**Issue by AviSarmani** [PyCQA/pylint#7682]
### Bug description

The code is:
```python
series = [1, 2, 3, 4, 5]
output_list = [
    (item, series[index])
    for index, item in enumerate(series, start=1)
    if index < len(series)
]
print(output_list)
```
Which produces the output:
`[(1, 2), (2, 3), (3, 4), (4, 5)]`

When there is a start parameter on the enumeration, "item" and "series[index]" are not the same object
but pylint suggests:
`R1736: Unnecessary list index lookup, use 'item' instead (unnecessary-list-index-lookup)`
If we follow the advice we would get a different result set 
`[(1, 1), (2, 2), (3, 3), (4, 4)]`

I think the error should be suppressed when there's a start index on the enumeration

### Command used

```shell
pylint .\pylint.py
```

### Pylint output

```shell
************* Module pylint
pylint.py:4:11: R1736: Unnecessary list index lookup, use 'item' instead (unnecessary-list-index-lookup)
```

### Expected behavior

No error at all

### Pylint version

```shell
pylint 2.14.1
astroid 2.11.5
Python 3.9.6 (tags/v3.9.6:db3ff76, Jun 28 2021, 15:26:21) [MSC v.1929 64 bit (AMD64)]
```

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue, I can reproduce that on the latest main.

**Comment by clavedeluna** [comment]
Also really similar to https://github.com/PyCQA/pylint/pull/7334/files so looks like this check is often getting false positives
